### PR TITLE
0.9.4 -> 0.9.5

### PR DESCRIPTION
**Pull request by pmlandwehr**

### CODE REVIEW

**Comment by ocefpaf** [comment]
@pmlandwehr the failure,

``` shell
    from .qt_compat import QtCore, QtGui, QtWidgets, _getSaveFileName, __version__
  File "/opt/conda/envs/_test/lib/python3.4/site-packages/matplotlib/backends/qt_compat.py", line 124, in <module>
    from PyQt4 import QtCore, QtGui
ImportError: libicui18n.so.56: cannot open shared object file: No such file or directory
TESTS FAILED: ggplot-0.9.5-py34_0
```

is not related to the update here.

Some Pythons were updated in `defaults` and they are compiled against `xz 5.2.*`, that pulls `libxml` from `defaults` and this version is not compiled with `icu`. Somehow `qt` from `defaults` is compiled with `icu`, but that is not in the metadata. Hence a mismatch and a broken stack.

``` shell
> conda execute scripts/list_deps.py defaults --package qt --platform linux-64     
Fetching package metadata: ....
qt 4.8.6 0 linux-64                : fontconfig >=2.11.1, freetype >=2.5.2, openssl
qt 4.8.6 1 linux-64                : fontconfig >=2.11.1, freetype >=2.5.2, openssl
qt 4.8.6 2 linux-64                : fontconfig >=2.11.1, freetype >=2.5.2, openssl
qt 4.8.6 3 linux-64                : fontconfig >=2.11.1, freetype >=2.5.2, openssl
qt 4.8.7 1 linux-64                : fontconfig >=2.11.1, freetype >=2.5.2, libpng 1.6*, openssl
qt 4.8.7 2 linux-64                : fontconfig >=2.11.1, freetype >=2.5.2, libpng 1.6*, openssl, zlib
qt 4.8.7 3 linux-64                : fontconfig >=2.11.1, freetype >=2.5.2, libpng 1.6*, openssl 1.0.2*, zlib 1.2.*
pyqt 4.10.4 py26_0 linux-64        : python 2.6*, qt 4.8.5, sip 4.15.5
pyqt 4.10.4 py27_0 linux-64        : python 2.7*, qt 4.8.5, sip 4.15.5
pyqt 4.10.4 py33_0 linux-64        : python 3.3*, qt 4.8.5, sip 4.15.5
pyqt 4.10.4 py34_0 linux-64        : python 3.4*, qt 4.8.5, sip 4.15.5
pyqt 4.11.3 py26_0 linux-64        : python 2.6*, qt 4.8.6, sip 4.16.5
pyqt 4.11.3 py26_1 linux-64        : python 2.6*, qt 4.8.6, sip 4.16.5
pyqt 4.11.3 py27_0 linux-64        : python 2.7*, qt 4.8.6, sip 4.16.5
pyqt 4.11.3 py27_1 linux-64        : python 2.7*, qt 4.8.6, sip 4.16.5
pyqt 4.11.3 py33_0 linux-64        : python 3.3*, qt 4.8.6, sip 4.16.5
pyqt 4.11.3 py33_1 linux-64        : python 3.3*, qt 4.8.6, sip 4.16.5
pyqt 4.11.3 py34_0 linux-64        : python 3.4*, qt 4.8.6, sip 4.16.5
pyqt 4.11.3 py34_1 linux-64        : python 3.4*, qt 4.8.6, sip 4.16.5
pyqt 4.11.3 py35_1 linux-64        : python 3.5*, qt 4.8.6, sip 4.16.5
pyqt 4.11.4 py27_0 linux-64        : python 2.7*, qt >=4.8.6, sip >=4.16.4
pyqt 4.11.4 py27_1 linux-64        : python 2.7*, qt >=4.8.7, sip >=4.16.4
pyqt 4.11.4 py27_2 linux-64        : python 2.7*, qt >=4.8.7, sip >=4.16.4
pyqt 4.11.4 py27_3 linux-64        : python 2.7*, qt >=4.8.7, sip >=4.16.4
pyqt 4.11.4 py34_0 linux-64        : python 3.4*, qt >=4.8.6, sip >=4.16.4
pyqt 4.11.4 py34_1 linux-64        : python 3.4*, qt >=4.8.7, sip >=4.16.4
pyqt 4.11.4 py34_2 linux-64        : python 3.4*, qt >=4.8.7, sip >=4.16.4
pyqt 4.11.4 py34_3 linux-64        : python 3.4*, qt >=4.8.7, sip >=4.16.4
pyqt 4.11.4 py35_0 linux-64        : python 3.5*, qt >=4.8.6, sip >=4.16.4
pyqt 4.11.4 py35_1 linux-64        : python 3.5*, qt >=4.8.7, sip >=4.16.4
pyqt 4.11.4 py35_2 linux-64        : python 3.5*, qt >=4.8.7, sip >=4.16.4
pyqt 4.11.4 py35_3 linux-64        : python 3.5*, qt >=4.8.7, sip >=4.16.4
```

 (Ping @pelson another place where the `xz` pinning would come in handy.)

**Comment by pelson** [comment]
> Ping @pelson another place where the xz pinning would come in handy

Agreed, but is this a case of bad metadata on defaults? If so, has this been raised?

**Comment by ocefpaf** [comment]
> Agreed, but is this a case of bad metadata on defaults? If so, has this been raised?

I think that is part of the problem but I could not find what part/dependency of `qt` causing it to link to `icu`.

Ping @ccordoba12 and @msarahan maybe you can help us figure this out.

**Comment by pelson** [comment]
I have tracked the issue down in https://github.com/conda-forge/matplotlib-feedstock/issues/36 (`defaults` doesn't reflect the icu dependency for `qt`).

EDIT: I looks like @ocefpaf had already drawn that conclusion. Sorry for duplicating your work @ocefpaf.

**Comment by pmlandwehr** [comment]
Well, `0.9.7` is out now. Ticking version again and rebuilding. Maybe everything upstream has been patched....?

**Comment by jakirkham** [comment]
It is weird as this issue doesn't crop up on Python 2.7 or 3.5 (tested locally), but does appear on Python 3.4.

**Comment by jakirkham** [comment]
I'm having trouble reproducing this problem with `qt` or `pyqt` alone. Maybe there is something else going on? Though it seems so clearly to be an issue with one of those two packages. 😕

**Comment by ocefpaf** [comment]
@pmlandwehr a workaround is to "force" install `icu` as if it were a run-time dependency for `ggplot`. Just add a comment that it is a workaround and should be removed once we fix our stack.

**Comment by pmlandwehr** [comment]
Booo. Maybe just skip that build?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -40,6 +40,8 @@ requirements:
     - scipy
     - six
     - statsmodels
+    - icu  # [win and py34]
```
Comment (ocefpaf) [inline recipe/meta.yaml]: Why `Win and py34`? The issue happens on Linux, right?

Comment (pmlandwehr) [inline recipe/meta.yaml]: Ah - my bad, then. Confused my CIs.

**Comment by pmlandwehr** [comment]
The py3k patch isn't related to the `icu` issue, but is going to be important to get in; see [here.](https://github.com/yhat/ggplot/commit/0e05ea562ac30071f123a11da4b18e80e307fc15) It should be done by `0.10.2`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -7,10 +9,13 @@ package:
 source:
   fn: ggplot-{{ version }}.tar.gz
   url: https://pypi.io/packages/source/g/ggplot/ggplot-{{ version }}.tar.gz
-  md5: 0fbaa3123cb2d4b060b200a106340421
+  {{ hash_type }}: {{ hash_val }}
+  patch:
```
Comment (ocefpaf) [inline recipe/meta.yaml]: I believe it is `patches`.

Comment (pmlandwehr) [inline recipe/meta.yaml]: GAH

**Comment by pmlandwehr** [comment]
Well, poop emoji.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -38,6 +43,8 @@ requirements:
     - scipy
     - six
     - statsmodels
+    - icu  # [linux and py34]
```
Comment (ocefpaf) [inline recipe/meta.yaml]: I believe the failure will happen in all Pythons and not only `PY34`. Maybe not `PY27`, but still it is OK to add the workaround across the board here.

**Comment by ocefpaf** [comment]
:tada:

**Comment by pmlandwehr** [comment]
yaaaaaay

**Pull request merged**